pkg/datarepair/datarepair_test.go

```diff
@@ -271,6 +284,134 @@ func TestRepairMultipleDisqualified(t *testing.T) {
 	})
 }
 
+// TestDataRepairUploadLimits does the following:
+// - Uploads test data to nodes
+// - Get one segment of that data to check in which nodes its pieces are stored
+// - Kills as many nodes as needed which store such segment pieces
+// - Triggers data repair
+// - Verify that the number of pieces which repaired has uploaded don't overpass
+//	 the established limit (success threshold + % of excess)
+func TestDataRepairUploadLimit(t *testing.T) {
```

Comment:
what does this test that is not tested inside `TestDataRepair`?

Reply:
`TestDataRepair` is specifically testing that repair does what it must do in terms that once a segment is repaired it can be downloaded without any issue and the downloaded segment after the repair has the nodes which received the repaired shares (this check is what it has to be added a back as you commented above).

This test specifically verifies that the repairer only upload up to a maximum number of nodes established by the optimal threshold and the maximum excess ratio.